delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resources/ResourceRejection.scala

```diff
@@ -133,24 +134,15 @@ object ResourceRejection {
   final case class WrappedProjectRejection(rejection: ProjectRejection) extends ResourceRejection(rejection.reason)
 
   /**
-    * Signals a rejection caused when interacting with the organizations API
-    */
-  final case class WrappedOrganizationRejection(rejection: OrganizationRejection)
-      extends ResourceRejection(rejection.reason)
-
-  /**
-    * Rejection returned when attempting to validate a resource against a schema that is deprecated.
-    *
-    * @param ref     the schema reference
+    * Signals a rejection caused when interacting with the schemas API
     */
-  final case class SchemaIsDeprecated(ref: ResourceRef) extends ResourceRejection(s"Schema '$ref' is deprecated.")
+  final case class WrappedSchemaRejection(rejection: SchemaRejection) extends ResourceRejection(rejection.reason)
 
   /**
-    * Rejection returned when attempting to validate a resource against a schema that is deprecated.
-    *
-    * @param ref     the schema reference
+    * Signals a rejection caused when interacting with the organizations API
     */
-  final case class SchemaNotFound(ref: ResourceRef) extends ResourceRejection(s"Schema '$ref' not found.")
+  final case class WrappedOrganizationRejection(rejection: OrganizationRejection)
+      extends ResourceRejection(rejection.reason)
```

Comment:
we could unify these with a type param:
```
final case class WrappedRejection[R <: Rejection](underlying: R)
    extends ResourceRejection(underlying.reason)
```

Reply:
I prefer it explicit in this way since looking at the rejections you know which ones are supported. If you have an [R] it tells you nothing. For pattern matching it would be also harder, to find the jsonLdEncoding of the rejection and other edge cases like this.